configs/components/openssl.rb

```diff
@@ -1,88 +1,215 @@
+    end
     pkg.build_requires 'pl-gcc'
+
+    if platform.name =~ /debian-8-arm/
+      pkg.build_requires 'xutils-dev'
+    end
   end
 
-  ## BUILD CONFIGURATION
-  install_prefix = "INSTALL_PREFIX=/" unless platform.is_windows?
+  #########
+  # PATCHES
+  #########
 
-  if platform.is_macos?
-    target = 'darwin64-x86_64-cc'
-    cflags = settings[:cflags]
-    ldflags = ''
-  elsif platform.is_windows?
+  if platform.is_windows?
     pkg.apply_patch 'resources/patches/openssl/openssl-1.0.0l-use-gcc-instead-of-makedepend.patch'
+    # This patch removes the option `-DOPENSSL_USE_APPLINK` from the mingw openssl congifure target
+    # This brings mingw more in line with what is happening with mingw64. All applink does it makes
+    # it possible to use the .dll compiled with one compiler with an application compiled with a
+    # different compiler. Given our openssl should only be interacting with things that we build,
+    # we can ensure everything is build with the same compiler.
     pkg.apply_patch 'resources/patches/openssl/openssl-mingw-do-not-build-applink.patch'
-    target = platform.architecture == "x64" ? "mingw64" : "mingw"
-    pkg.environment "PATH", "$(shell cygpath -u #{settings[:gcc_bindir]}):$(PATH)"
-    pkg.environment "CYGWIN", settings[:cygwin]
-    pkg.environment "CC", settings[:cc]
-    pkg.environment "CXX", settings[:cxx]
-    pkg.environment "MAKE", platform[:make]
-    cflags = settings[:cflags]
-    ldflags = settings[:ldflags]
-  else # Linux Platforms
-    pkg.environment "PATH", "/opt/pl-build-tools/bin:$(PATH):/usr/local/bin"
-    target = 'linux-x86_64'
-    cflags = settings[:cflags]
-    ldflags = "#{settings[:ldflags]} -Wl,-z,relro"
+  elsif platform.is_aix?
+    pkg.apply_patch 'resources/patches/openssl/add-shell-to-engines_makefile.patch'
+    pkg.apply_patch 'resources/patches/openssl/openssl-1.0.0l-use-gcc-instead-of-makedepend.patch'
+  elsif platform.is_solaris?
+    if platform.os_version == '10'
+      pkg.apply_patch 'resources/patches/openssl/solaris-10-domd-shell-compatability-fix.patch'
+    end
+    pkg.apply_patch 'resources/patches/openssl/add-shell-to-engines_makefile.patch'
+    pkg.apply_patch 'resources/patches/openssl/openssl-1.0.0l-use-gcc-instead-of-makedepend.patch'
+  elsif platform.is_linux?
+    if platform.name =~ /debian-8-arm/
+      pkg.apply_patch 'resources/patches/openssl/openssl-1.0.0l-use-gcc-instead-of-makedepend.patch'
+    end
+  end
+
+  ###########
+  # CONFIGURE
+  ###########
+
+  # OpenSSL Configure doesn't honor CFLAGS or LDFLAGS as environment variables.
+  # Instead, those should be passed to Configure at the end of its options, as
+  # any unrecognized options are passed straight through to ${CC}. Defining
+  # --libdir ensures that we avoid the multilib (lib/ vs. lib64/) problem,
+  # since configure uses the existence of a lib64 directory to determine
+  # if it should install its own libs into a multilib dir. Yay OpenSSL!
```

Comment:
@jsane when you have a chance, would you please review the openssl configuration?

Reply:
Yes I had a question earlier in the discussion somewhere. 

Besides, I was also planning to remove 3DES separately (to address  https://www.openssl.org/blog/blog/2016/08/24/sweet32/) at least for agent. After doing some experimentation found that one also need to disable SRP, PSK for 3DES to be completely removed. Since this is going to get tested may I request to also include this change (no-3des no-srp no-psk). thx